src/anomalib/utils/metrics/perimg/plot.py

```diff
@@ -474,3 +474,328 @@ def _sort_legend(handles: list, labels: list[str]):
     ax.set_title("Per-Image Overlap Curves (AUC boxplot statistics)")
 
     return fig, ax
+
+
+def _add_integration_range_to_pimo_curves(
+    ax: Axes,
+    bounds: tuple[float | None, float | None] = (None, None),
+):
+    """TODO docstring"""
+    current_legend = ax.get_legend()
+
+    if len(bounds) != 2:
+        raise ValueError(f"Expected argument `bounds` to be a tuple of size 2, but got size {len(bounds)}.")
+
+    lbound, ubound = bounds
+
+    if lbound is not None:
+        _validate_nonzero_rate(lbound)
+
+    if ubound is not None:
+        _validate_nonzero_rate(ubound)
+
+    if lbound is not None and ubound is not None and lbound >= ubound:
+        raise ValueError(
+            f"Expected argument `bounds` to be (lbound, ubound), such that lbound < ubound, but got {bounds}."
+        )
+
+    handles = [
+        ax.axvspan(
+            lbound if lbound is not None else 0,
+            ubound if ubound is not None else 1,
+            label="FPR Interval",
+            color="cyan",
+            alpha=0.2,
+        ),
+    ]
+
+    if ubound is not None:
+        handles.append(
+            ax.axvline(
+                ubound,
+                label=f"Upper Bound ({float(100 * ubound):.2g}%)",
+                linestyle="--",
+                color="black",
+            )
+        )
+
+    if lbound is not None:
+        handles.append(
+            ax.axvline(
+                lbound,
+                label=f"Lower Bound ({float(100 * lbound):.2g}%)",
+                linestyle="--",
+                color="gray",
+            )
+        )
+
+    ax.legend(
+        handles,
+        [h.get_label() for h in handles],
+        title="AUC integration",
+        loc="center right",
+        fontsize="small",
+        title_fontsize="small",
+    )
+
+    if current_legend is not None:
+        ax.add_artist(current_legend)
+
+
+# =========================================== PImFPR ===========================================
+
+
+def plot_pimfpr_curves_norm_vs_anom(
```

Comment:
Also, would it be possible to have a function in which norm or norm-vs-anom could be an argument that the user sets? Like `plot_pimfpr_curves(..., normal_only=True` or something like that

Reply:
> What does `pim` stand for here?

`per-image`

> Also, would it be possible to have a function in which norm or norm-vs-anom could be an argument that the user sets? Like `plot_pimfpr_curves(..., normal_only=True` or something like that

I was exactly avoiding that to not encourage put everything in one :sweat_smile: . They're secondary functions, not an actual endpoint, so i prefered to keep them simpler and doing their own thing to avoid (eventual) interface complications.